src/System.Text.Json/src/System/Text/Json/JsonParseObject.cs

```diff
@@ -0,0 +1,350 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Text.Utf8;
+
+namespace System.Text.Json
+{
+    public struct JsonParseObject
+    {
+        private byte[] _buffer;
+        private int _start;
+        private int _end;
+
+        private const int RowSize = 9;
+
+        public JsonParseObject(byte[] buffer, int start, int end)
```

Comment:
I think the first parameter has to be UTF8 encoded text? If that's correct, we should rename the parameter to utf8JsonText or something like that.